src/emit.js

```diff
@@ -8,10 +8,27 @@ const defs: EventOptions = {
   composed: false
 };
 
+const hasNativeSupport = verifyNativeEventSupport();
+
+function verifyNativeEventSupport () {
+  try {
+    const detail = {};
+    const test = new CustomEvent('testEvent', { composed: true, detail });
+    return test.composed === true && test.detail === detail;
+  } catch (error) {
+    return false;
+  }
+}
+
 export function emit (elem: HTMLElement, name: string, opts: EventOptions): boolean {
   opts = { ...defs, ...opts };
-  const e: ComposedCustomEvent = document.createEvent('CustomEvent');
-  e.initCustomEvent(name, opts.bubbles, opts.cancelable, opts.detail);
-  Object.defineProperty(e, 'composed', { value: opts.composed });
+  let e: ComposedCustomEvent;
+  if (hasNativeSupport) {
```

Comment:
Instead of checking `hasNativeSupport` and using the function above, you can probably just do `'composed' in CustomEvent.prototype`.